src/System.Reflection.Metadata/tests/PortableExecutable/DebugDirectoryTests.cs

```diff
@@ -153,10 +153,12 @@ private void ValidateEmbeddedPortablePdb(PEReader reader)
             Assert.Equal(DebugDirectoryEntryType.Reproducible, entries[1].Type);
             Assert.Equal(DebugDirectoryEntryType.EmbeddedPortablePdb, entries[2].Type);
 
-            var provider = reader.ReadEmbeddedPortablePdbDebugDirectoryData(entries[2]);
-            var pdbReader = provider.GetMetadataReader();
-            var document = pdbReader.GetDocument(pdbReader.Documents.First());
-            Assert.Equal(@"C:\Documents.cs", pdbReader.GetString(document.Name));
+            using (var provider = reader.ReadEmbeddedPortablePdbDebugDirectoryData(entries[2]))
+            {
+                var pdbReader = provider.GetMetadataReader();
```

Comment:
I understand that. The provider keeps the memory alive, whether it's managed or native, right? Shouldn't the reader keep the provider referenced to keep it alive?